Formula/hyx.rb

```diff
@@ -0,0 +1,22 @@
+class Hyx < Formula
+  desc "Powerful hex editor for the console"
+  homepage "https://yx7.cc/code/"
+  url "https://yx7.cc/code/hyx/hyx-2021.06.09.tar.xz"
+  sha256 "8d4f14e58584d6cc8f04e43ca38042eed218882a389249c20b086730256da5eb"
+  license "MIT"
+
+  def install
+    inreplace "Makefile", " -Wl,-s", " "
+    inreplace "Makefile", " -Wl,-z,relro,-z,now -fpic -pie", " "
+    inreplace "Makefile", " -Werror", " "
+
+    system "make"
+
+    bin.install "hyx"
+    doc.install "license.txt"
+  end
+
+  test do
+    assert_match "This is hyx", shell_output("#{bin}/hyx -v").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
For this case, the version test is all there is.  I will need to make a test that uses `expect` and update tonight.

Reply:
Fixed in https://github.com/Homebrew/homebrew-core/pull/106085/commits/6405e291d009b3d735399ad629515439990d4fa4.